fpsd/features.py

```diff
@@ -0,0 +1,747 @@
+#!/usr/bin/env python3.5
+import pandas as pd
+import os
+from sqlalchemy import create_engine
+from tqdm import tqdm
+import subprocess
+
+
+def compute_bursts(df):
+    """Compute list of burst lengths for a given trace
+    """
+    bursts = []
+    current_burst_length = 0
+
+    for row in df.itertuples():
+        if row.ingoing == False:
+            current_burst_length += 1
+        elif row.ingoing == True and current_burst_length > 0:
+            bursts.append(current_burst_length)
+            # Reset burst length
+            current_burst_length = 0
+
+    # If we were in a burst at the end of the trace, let's save it
+    if current_burst_length > 0:
+        bursts.append(current_burst_length)
+
+    # For some features we need to know the ordering of bursts,
+    # so let's save this information as well
+    ranks = list(range(1, len(bursts) + 1))
+    return bursts, ranks
+
+
+class FeatureStorage():
+    def __init__(self, test_db=False):
+        """Set up database engine"""
+        self.TEST_MODE = test_db
```

Comment:
I think that testing logic should be separate from our classes. If we create a test db that mirrors the real one, and set `PGDATABSE=fpsd` in our test fixtures, that should obviate the need for this.